src/hotspot/cpu/ppc/frame_ppc.cpp

```diff
@@ -324,7 +324,11 @@ bool frame::is_interpreted_frame_valid(JavaThread* thread) const {
 
   // first the method
 
-  Method* m = *interpreter_frame_method_addr();
+  Method** m_addr = interpreter_frame_method_addr();
+  if (m_addr == nullptr || !os::is_readable_pointer(m_addr)) {
+    return false;
+  }
+  Method* m = *m_addr;
 
```

Comment:
Wouldn't it make more sense to define a function which takes a pointer to a (possible) method pointer and returns true if the method is valid?

Reply:
It's not clear to me how the proposed solution could make less sense than another solution - could you elaborate please? Are you concerned about code repetition? I must say this code is already rather repetitive across the different architectures and, while I don't want to make it any worse in that respect, I'm trying to make the smallest possible change to prevent the observed crash from recurring.